flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/BufferConsumer.java

```diff
@@ -44,42 +44,58 @@
 
 	private int currentReaderPosition;
 
+	/** Whether the underlying {@link Buffer} can be shared by multi {@link BufferConsumer} instances. */
+	private final boolean isShareable;
```

Comment:
`{@link Buffer}` here is not very accurate because we have two cases for tagging this field as true.

- `bufferBuilder.createBufferConsumer(true)`: in this case, every created `BufferConsumer` refers to different `Buffer` instances

- `BufferConsumer#copy()`:  only in this case the underlying `Buffer` is shared by other instances.

So maybe we adjust the description as `Whether the respective writable {@link BufferBuilder} is shared .....`

Reply:
By tag the BufferConsumer as shareable, I mean underlying buffer can be in the future, not the buffer is shared immediately.